include/wlr/types/wlr_buffer.h

```diff
@@ -24,6 +24,8 @@ struct wlr_shm_attributes {
 };
 
 struct wlr_buffer_impl {
+	bool (*is_instance)(struct wl_resource *resource);
+	struct wlr_buffer *(*create)(struct wl_resource *resource);
```

Comment:
I'd prefer to introduce a separate struct for these, e.g. `wlr_wl_buffer_impl`. `wlr_buffer` itself is completely separate from the Wayland protocol and `wl_resource`.

Reply:
I'd prefer to name this `buffer_from_resource` rather than `create`, for consistency with `wlr_buffer_from_resource`.

Reply:
I was hesitating between the two and chose `create` to mimic [`shm_client_buffer_create`](https://github.com/swaywm/wlroots/blob/master/types/wlr_buffer.c#L167), will make the changes.